src/transformStateWithClones.js

```diff
@@ -7,7 +7,31 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateClonesArray = [];
+  const stateCopyObject = { ...state };
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(stateCopyObject, action.extraData);
+        stateClonesArray.push({ ...stateCopyObject });
+        break;
+      case 'removeProperties':
+        action.keysToRemove.forEach((key) => delete stateCopyObject[key]);
+        stateClonesArray.push({ ...stateCopyObject });
+        break;
+      case 'clear':
+        Object.keys(stateCopyObject).forEach(
+          (key) => delete stateCopyObject[key]
+        );
+        stateClonesArray.push({ ...stateCopyObject });
+        break;
+      default:
+        return 'Error: invalid value received';
```

Comment:
The function should not return a string error message in case of invalid value received. It's better to throw an error or simply ignore the invalid action.